src/mlpack/methods/kernel_svm/kernel_svm.hpp

```diff
@@ -0,0 +1,224 @@
+/**
+ * @file methods/kernel_svm/kernel_svm.hpp
+ * @author Himanshu Pathak
+ *
+ * An implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <ensmallen.hpp>
+#include <string>
+
+
+namespace mlpack {
+namespace svm {
+
+/**
+ * The KernelSVM class implements an smo algorithm for support vector machine
+ * model, and supports training with multiple non-linear and linear kenel.
+ * The class supports different observation types via the MatType template
+ * parameter; for instance, support vector classification can be performed
+ * on sparse datasets by specifying arma::sp_mat as the MatType parameter.
+ *
+ *
+ * @code
+ * @inproceedings{CS 229, Autumn 2009,
+ * title        = {The Simplified SMO Algorithm.},
+ * year         = {2009}
+ * }
+ * @endcode
```

Comment:
Can you complete this citation? Also is this a research paper?

Reply:
Sure will change it in next commit.